### ISSUE TITLE
txt2json

### ISSUE THREAD

**Issue by linchunmian** [jwchoi384/Gaussian_YOLOv3#3]
Hi, could you please provide script for us that converts txt to json format, so that we can evaluate on bdd-data? My ground-truth file is txt format due to being used for yolov3 model training.
Thanks a lot.

**Comment by jwchoi384** [comment]
@ChunmianLin 
Hi, I already uplaoded GT file for validation set ("gt_bdd_val.json") in "bdd_evaluation" directory.
If you follow the readme in my repo, you can evaluate BDD validation set.

Do you need a code for converting txt to json on the training set instead of the validation set?
I upload "convert_txt_to_bdd_eval_json.py" file in bdd_evaluation directory.
And follow the steps:
1. Please install pillow image library
2. Setup the mypath in the code ("convert_txt_to_bdd_eval_json.py") as directory that contains yolo format text file. 
*For example, if you unzip the bdd dataset on /hdd/ directory, set the mypath as “/hdd/bdd100k/labels/100k/train_yolo/”.*
3. Execute the python code, then you get the gt.json file
`python convert_txt_to_bdd_eval_json.py`

**Comment by linchunmian** [comment]
Thanks a lot.  I will recheck it tonight.Sincerely!Chunmian Lin发自我的iPhone------------------ Original ------------------From: Jiwoong Choi  <notifications@github.com>Date: Fri,Oct 25,2019 3:58 PMTo: jwchoi384/Gaussian_YOLOv3 <Gaussian_YOLOv3@noreply.github.com>Cc: ChunmianLin <cmlin@buaa.edu.cn>, Mention <mention@noreply.github.com>Subject: Re: [jwchoi384/Gaussian_YOLOv3] txt2json (#3)@ChunmianLin

Hi, I already uplaoded GT file for validation set ("gt_bdd_val.json") in "bdd_evaluation" directory.

If you follow the readme in my repo, you can evaluate BDD validation set.

Do you need a code for converting txt to json on the training set instead of the validation set?

I upload "convert_txt_to_bdd_eval_json.py" file in bdd_evaluation directory.

And follow the steps:



Please install pillow image library

Setup the mypath in the code ("convert_txt_to_bdd_eval_json.py") as directory that contains yolo format text file.

For example, if you unzip the bdd dataset on /hdd/ directory, set the mypath as “/hdd/bdd100k/labels/100k/train_yolo/”.

Execute the python code, then you get the gt.json file

python convert_txt_to_bdd_eval_json.py





—You are receiving this because you were mentioned.Reply to this email directly, view it on GitHub, or unsubscribe.

[

{

"@context": "http://schema.org",

"@type": "EmailMessage",

"potentialAction": {

"@type": "ViewAction",

"target": "https://github.com/jwchoi384/Gaussian_YOLOv3/issues/3?email_source=notifications\u0026email_token=AMWSBRYIXNRBAESNX6FBPTTQQKRJZA5CNFSM4JE5KLS2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOECHRCEQ#issuecomment-546246930",

"url": "https://github.com/jwchoi384/Gaussian_YOLOv3/issues/3?email_source=notifications\u0026email_token=AMWSBRYIXNRBAESNX6FBPTTQQKRJZA5CNFSM4JE5KLS2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOECHRCEQ#issuecomment-546246930",

"name": "View Issue"

},

"description": "View this Issue on GitHub",

"publisher": {

"@type": "Organization",

"name": "GitHub",

"url": "https://github.com"

}

}

]

**Comment by linchunmian** [comment]
Hi, could you please provide json2txt script of bds dataset for me? I found the format of data is different due to I use the old label that each sample in training and val set has its individual json file. Thanks a lot.Chunmian Lin发自我的iPhone------------------ Original ------------------From: Jiwoong Choi  <notifications@github.com>Date: Fri,Oct 25,2019 3:58 PMTo: jwchoi384/Gaussian_YOLOv3 <Gaussian_YOLOv3@noreply.github.com>Cc: ChunmianLin <cmlin@buaa.edu.cn>, Mention <mention@noreply.github.com>Subject: Re: [jwchoi384/Gaussian_YOLOv3] txt2json (#3)@ChunmianLin

Hi, I already uplaoded GT file for validation set ("gt_bdd_val.json") in "bdd_evaluation" directory.

If you follow the readme in my repo, you can evaluate BDD validation set.

Do you need a code for converting txt to json on the training set instead of the validation set?

I upload "convert_txt_to_bdd_eval_json.py" file in bdd_evaluation directory.

And follow the steps:



Please install pillow image library

Setup the mypath in the code ("convert_txt_to_bdd_eval_json.py") as directory that contains yolo format text file.

For example, if you unzip the bdd dataset on /hdd/ directory, set the mypath as “/hdd/bdd100k/labels/100k/train_yolo/”.

Execute the python code, then you get the gt.json file

python convert_txt_to_bdd_eval_json.py





—You are receiving this because you were mentioned.Reply to this email directly, view it on GitHub, or unsubscribe.

[

{

"@context": "http://schema.org",

"@type": "EmailMessage",

"potentialAction": {

"@type": "ViewAction",

"target": "https://github.com/jwchoi384/Gaussian_YOLOv3/issues/3?email_source=notifications\u0026email_token=AMWSBRYIXNRBAESNX6FBPTTQQKRJZA5CNFSM4JE5KLS2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOECHRCEQ#issuecomment-546246930",

"url": "https://github.com/jwchoi384/Gaussian_YOLOv3/issues/3?email_source=notifications\u0026email_token=AMWSBRYIXNRBAESNX6FBPTTQQKRJZA5CNFSM4JE5KLS2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOECHRCEQ#issuecomment-546246930",

"name": "View Issue"

},

"description": "View this Issue on GitHub",

"publisher": {

"@type": "Organization",

"name": "GitHub",

"url": "https://github.com"

}

}

]

**Comment by jwchoi384** [comment]
@ChunmianLin 
Hi, I uplaoded the file (json to txt, "bdd_json2txt") in "bdd_evaluation" directory.

1. Setup the mypath in "bdd_json2txt.py" as directory that contains bdd json file. 
2. Setup the outpath "bdd_json2txt.py" as directory that plan to save txt file. 
3. python bdd_json2txt.py

**Comment by linchunmian** [comment]
Thanks!Sincerely!发自我的iPhone------------------ Original ------------------From: Jiwoong Choi  <notifications@github.com>Date: Mon,Nov 4,2019 5:50 PMTo: jwchoi384/Gaussian_YOLOv3 <Gaussian_YOLOv3@noreply.github.com>Cc: ChunmianLin <cmlin@buaa.edu.cn>, Mention <mention@noreply.github.com>Subject: Re: [jwchoi384/Gaussian_YOLOv3] txt2json (#3)@ChunmianLin
Hi, I uplaoded the file (json to txt, "bdd_json2txt") in ("gt_bdd_val.json") in "bdd_evaluation" directory.

Setup the mypath in "bdd_json2txt.py" as directory that contains bdd json file.
Setup the outpath "bdd_json2txt.py" as directory that plan to save txt file.
python bdd_json2txt.py


—You are receiving this because you were mentioned.Reply to this email directly, view it on GitHub, or unsubscribe.
[
{
"@context": "http://schema.org",
"@type": "EmailMessage",
"potentialAction": {
"@type": "ViewAction",
"target": "https://github.com/jwchoi384/Gaussian_YOLOv3/issues/3?email_source=notifications\u0026email_token=AMWSBR3AGZQV3LVELI6RLVDQR7V5JA5CNFSM4JE5KLS2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEC6WA7Y#issuecomment-549281919",
"url": "https://github.com/jwchoi384/Gaussian_YOLOv3/issues/3?email_source=notifications\u0026email_token=AMWSBR3AGZQV3LVELI6RLVDQR7V5JA5CNFSM4JE5KLS2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEC6WA7Y#issuecomment-549281919",
"name": "View Issue"
},
"description": "View this Issue on GitHub",
"publisher": {
"@type": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]
